Sources/Purchasing/StoreKit2/StoreKit2Receipt.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKit2Receipt.swift
+//
+//  Created by MarkVillacampa on 26/10/23.
+
+import StoreKit
+
+/// A type that resembles the structure of a StoreKit 1 receipt using StoreKit 2 data.
+struct StoreKit2Receipt: Equatable {
+
+    struct SubscriptionState: RawRepresentable, Equatable {
```

Comment:
Why not use a regular enum?

Reply:
This wraps a [RenewalState](https://developer.apple.com/documentation/storekit/product/subscriptioninfo/renewalstate/) which is itself a rawrepresentable int struct.